icu4c/source/i18n/unitsdata.cpp

```diff
@@ -0,0 +1,194 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+
+#include <utility>
+
+#include "cstring.h"
+#include "resource.h"
+#include "unitsdata.h"
+#include "uresimp.h"
+
+U_NAMESPACE_BEGIN
+
+namespace {
+
+/**
+ * A ResourceSink that collects conversion rate information.
+ *
+ * This class is for use by ures_getAllItemsWithFallback. Example code for
+ * collecting conversion info for "mile" and "foot" into conversionInfoOutput:
+ *
+ *     UErrorCode status = U_ZERO_ERROR;
+ *     ures_getByKey(unitsBundle, "convertUnits", &fillIn, &status);
+ *     MaybeStackVector<ConversionRateInfo> conversionInfoOutput;
+ *     ConversionRateDataSink convertSink(conversionInfoOutput);
+ *     ures_getAllItemsWithFallback(fillIn, "mile", convertSink, status);
+ *     ures_getAllItemsWithFallback(fillIn, "foot", convertSink, status);
+ */
+class ConversionRateDataSink : public ResourceSink {
+  public:
+    /**
+     * Constructor.
+     * @param out The vector to which ConversionRateInfo instances are to be
+     * added.
+     */
+    explicit ConversionRateDataSink(MaybeStackVector<ConversionRateInfo> &out) : outVector(out) {}
```

Comment:
You pass around `MaybeStackVector<ConversionRateInfo> &` a lot. I would make a first-class type called ConversionRateInfoList with the methods you need, that contains the MaybeStackVector inside.

Reply:
This is related to Hugo pull request: #31

Reply:
I haven't done this yet - I'm transferring this feedback to #31 so we can continue there. Need to coordinate with Younies, at the moment `getConversionRatesInfo(const MaybeStackVector<MeasureUnit> &, UErrorCode &status)` is the interface between his code and mine (which needs to change, to "getAllConversionRates" perhaps with a new return type).

Reply:
Thread can continue here:
https://github.com/sffc/icu/pull/31/commits/2deeb28f3c51aa16e351e6c70ea547cb5cef6679#r408191824

Marking this one as resolved.